libs/types/src/models/submission/validator.class.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  SimpleValidationResult,
+  ValidationMessage,
+  ValidationMessages,
+} from './validation-result.type';
+import { IWebsiteFormFields } from './website-form-fields.interface';
+
+type KeysToOmit =
+  | 'mergeDefaults'
+  | 'getFormFieldFor'
+  | 'getFormFields'
+  | 'getProcessedTags';
+
+type ValidationArray<Fields extends IWebsiteFormFields> = ValidationMessage<
```

Comment:
nit: Ensure this is used elsewhere in the codebase where validations are done currently (i.e. itaku) to keep standards if this is to be the way forward.